routes/booking.js

```diff
@@ -0,0 +1,170 @@
+require("express-async-errors");
+const express = require("express");
+const { Booking } = require("../models/booking");
+const { Room } = require("../models/room");
+const isClient = require("../middleware/isClient");
+const isManager = require("../middleware/isManager");
+const router = express.Router();
+
+router.post("/requestToBook/", isClient, async (req, res) => {
+  const { arrivalDate, departureDate, roomId } = req.query;
+  const { user } = req;
+
+  if (!arrivalDate || !departureDate || !roomId || !user._id)
+    return res.status(400).send(`Invalid Request`);
+
+  const existingBookingBody = {
+    arrivalDate: { $gte: arrivalDate },
+    departureDate: { $lt: departureDate },
+    roomId,
+  };
+  const existingBooking = await Booking.findOne(existingBookingBody);
+
+  const room = await Room.findOne({ _id: roomId });
+  if (!room) res.status(400).send(`The room does not exists !`);
+
+  if (existingBooking && existingBooking.isApproved)
```

Comment:
I would suggest you keep these conditions separated from one another. Maybe firstly you could check if the room exists and return the error accordingly. And the other error which checks if a room is approved within the provided time frame.